ui_components/src/lib/components/input/InputComponent.svelte

```diff
@@ -136,24 +141,31 @@
         endEdit();
     }
 
-
+    // Input label
+    let title = readonly ? 'This field cannot be edited' : 'Click to edit field';
 </script>
 
 {#snippet enforceMaxChar()}
     <label class="form-label">
         Maximum characters
-        <input type="number" class="form-control" bind:value={config.maxNumChar}/>
+        <input type="number" class="form-control" bind:value={config.maxNumChar} readonly={readonly}/>
     </label>
 
     <div class="form-check form-switch">
         <label class="form-label">
             Enforce maximum characters
             <input class="form-check-input" type="checkbox" role="switch"
-                   bind:checked={config.enforceValueConstraints}>
+                   bind:checked={config.enforceValueConstraints} readonly={readonly} disabled={readonly}>
         </label>
     </div>
 {/snippet}
 
+{#snippet readOnlyBadge()}
+<span class="badge badge-secondary text-bg-secondary"
+  title="This is a standard field and cannot be modified.">
```

Comment:
Just a bit more feedback to the user that they can hide the field if they're not using it:

```suggestion
  title="This is a standard field and cannot be modified. You can hide this field by clicking the Disable option below.">
```